packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py

```diff
@@ -120,115 +75,32 @@ def mech_response(self) -> MechInteractionResponse:
-            request_id=self.request_id,
-            from_block=self.from_block,
-            timeout=self.params.contract_timeout,
-        )
-
-        if result:
-            self.set_mech_response_specs()
-
-        return result
-
-    def _handle_response(
-        self,
-        res: Optional[str],
-    ) -> Optional[Any]:
-        """Handle the response from the IPFS.
-
-        :param res: the response to handle.
-        :return: the response's result, using the given keys. `None` if response is `None` (has failed).
-        """
-        if res is None:
-            msg = f"Could not get the mech's response from {self.mech_response_api.api_id}"
-            self.context.logger.error(msg)
-            self.mech_response_api.increment_retries()
-            return None
-
-        self.context.logger.info(f"Retrieved the mech's response: {res}.")
-        self.mech_response_api.reset_retries()
-        return res
-
-    def _get_response(self) -> WaitableConditionType:
-        """Get the response data from IPFS."""
-        specs = self.mech_response_api.get_spec()
-        res_raw = yield from self.get_http_response(**specs)
-        res = self.mech_response_api.process_response(res_raw)
-        res = self._handle_response(res)
-
-        if self.mech_response_api.is_retries_exceeded():
-            error = "Retries were exceeded while trying to get the mech's response."
+    def _get_response(self) -> None:
+        """Get the response data."""
+        mech_responses = self.synchronized_data.mech_responses
+        if not mech_responses:
+            error = "No Mech responses in synchronized_data."
             self._mech_response = MechInteractionResponse(error=error)
-            return True
 
-        if res is None:
-            return False
-
-        try:
-            self._mech_response = MechInteractionResponse(**res)
-        except (ValueError, TypeError, KeyError):
-            self._mech_response = MechInteractionResponse.incorrect_format(res)
-
-        return True
+        self._mech_response = mech_responses[0]
 
     def _get_decision(
         self,
-    ) -> Generator[None, None, Tuple[Optional[int], Optional[float], Optional[float]]]:
+    ) -> Tuple[Optional[int], Optional[float], Optional[float]]:
         """Get vote, win probability and confidence."""
-        for step in (
-            self._get_block_number,
-            self._get_request_id,
-            self._get_response_hash,
-            self._get_response,
-        ):
-            yield from self.wait_for_condition_with_sleep(step)
-
+        self._get_response()
         self.context.logger.info(f"Decision has been received:\n{self.mech_response}")
         if self.mech_response.result is None:
             self.context.logger.error(
                 f"There was an error on the mech's response: {self.mech_response.error}"
             )
             return None, None, None
 
+        result = PredictionResponse(**json.loads(self.mech_response.result))
```

Comment:
This needs exception handling in case of a JSON decode error.

Reply:
Addressed